src/sage/rings/polynomial/multi_polynomial_element.py

```diff
@@ -668,11 +668,22 @@ def degree(self, x=None, std_grading=False):
             sage: R.<x,y> = GF(3037000453)[]                                            # needs sage.rings.finite_rings
             sage: R.zero().degree(x)
             -1
+
+        Ensure that :issue:`37603` is fixed::
+
+            sage: R.<x,y,z> = PolynomialRing(QQbar)
+            sage: f = 3*x^2 - 2*y + 7*x^2*y^2 + 5
+            sage: type(f.degree())
+            <class 'sage.rings.integer.Integer'>
+            sage: type(f.degree(x))
+            <class 'sage.rings.integer.Integer'>
+            sage: type(f.degree(x)) == type(f.degree(y)) == type(f.degree(z))
+            True
         """
         if x is None:
             if std_grading or not self.parent().term_order().is_weighted_degree_order():
-                return self.element().degree(None)
-            return self.weighted_degree(self.parent().term_order().weights())
+                return Integer(self.element().degree(None))
+            return Integer(self.weighted_degree(self.parent().term_order().weights()))
```

Comment:
This line is not necessary since [weighted_degree](https://github.com/sagemath/sage/blob/develop/src/sage/rings/polynomial/multi_polynomial.pyx#L2079) is wrapped.

Reply:
Thank you